src/SignalR/clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts

```diff
@@ -217,31 +216,43 @@ export class MessagePackHubProtocol implements IHubProtocol {
     private writeInvocation(invocationMessage: InvocationMessage): ArrayBuffer {
         const msgpack = msgpack5();
         const payload = msgpack.encode([MessageType.Invocation, invocationMessage.headers || {}, invocationMessage.invocationId || null,
-        invocationMessage.target, invocationMessage.arguments]);
+        invocationMessage.target, invocationMessage.arguments, invocationMessage.streamIds]);
 
         return BinaryMessageFormat.write(payload.slice());
     }
 
     private writeStreamInvocation(streamInvocationMessage: StreamInvocationMessage): ArrayBuffer {
         const msgpack = msgpack5();
         const payload = msgpack.encode([MessageType.StreamInvocation, streamInvocationMessage.headers || {}, streamInvocationMessage.invocationId,
-        streamInvocationMessage.target, streamInvocationMessage.arguments]);
+        streamInvocationMessage.target, streamInvocationMessage.arguments, streamInvocationMessage.streamIds]);
 
         return BinaryMessageFormat.write(payload.slice());
     }
 
-    private writeStreamData(streamDataMessage: StreamDataMessage): ArrayBuffer {
+    private writeStreamItem(streamItemMessage: StreamItemMessage): ArrayBuffer {
         const msgpack = msgpack5();
-        const payload = msgpack.encode([MessageType.StreamData, streamDataMessage.streamId,
-            streamDataMessage.item]);
+        const payload = msgpack.encode([MessageType.StreamItem, streamItemMessage.headers || {}, streamItemMessage.invocationId,
+        streamItemMessage.item]);
 
         return BinaryMessageFormat.write(payload.slice());
     }
 
-    private writeStreamComplete(streamCompleteMessage: StreamCompleteMessage): ArrayBuffer {
+    private writeCompletion(completionMessage: CompletionMessage): ArrayBuffer {
         const msgpack = msgpack5();
-        const payload = msgpack.encode([MessageType.StreamComplete, streamCompleteMessage.streamId,
-            streamCompleteMessage.error || null]);
+        const resultKind = completionMessage.error ? 1 : completionMessage.result ? 3 : 2;
```

Comment:
Are these values being used in multiple places? For example, both for writing the completionMessage and reading it? Might be worth making them constants.